### PR TITLE
feat(android): Android web-based test harness 🎢

### PR DESCRIPTION
**Pull request by mcdurdin**
This small harness simulates the Android app. I used it to try and dig deeper into the interactions with selected text in #5853.

It is very rough, but with some extra effort we could use this relatively easily for some low-level testing of KeymanWeb embedded integration.

@keymanapp-test-bot skip

### CODE REVIEW

File: web/testing/android-harness/build.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+THIS_DIR="$(dirname "$THIS_SCRIPT")"
+
+mkdir -p "$THIS_DIR/host"
+cp -R "$KEYMAN_ROOT/android/KMEA/app/src/main/assets/"* "$THIS_DIR/host/"
+cp "$KEYMAN_ROOT/web/release/unminified/embedded/keyman.js" "$THIS_DIR/host/keymanandroid.js"
+cp "$KEYMAN_ROOT/web/release/unminified/embedded/keyman.js.map" "$THIS_DIR/host/keyman.js.map"
+
+# Test keyboards
+
+cp "$KEYMAN_ROOT/web/testing/web_context_tests.js" "$THIS_DIR/host/"
```
Comment (darcywong00) [inline web/testing/android-harness/build.sh:20]: Is `web_context_tests.js` from a separate PR?

Comment (mcdurdin) [reply web/testing/android-harness/build.sh:20]: Yeah .... fixing this now

**Review by darcywong00 — APPROVED** [review]
lgtm

**Comment by darcywong00** [comment]
I've already approved. But when I ran this in the follow-on PR, I'm getting console exceptions:

```
Uncaught DOMException: Blocked a frame with origin "null" from accessing a cross-origin frame.
    at file:///C:/src/keymanapp/web/testing/android-harness/host/android-host.js:4:35
```

and a big black box next to the text area.

**Comment by mcdurdin** [comment]
When running this, you need to run in Chrome in mobile emulation mode, to enable touch. It's a bit shaky because KMW embedded assumes that touch is available, and so you'll see errors like this one if not in mobile emulation.

File: common/test/keyboards/web_context_tests/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+==============
+
+Version 1.0
+
+Description
+-----------
+web_context_tests is used for testing selection rules. It has a single rule:
+
+```
+  'abc' + 'd' > '!'
```
Comment (darcywong00) [inline common/test/keyboards/web_context_tests/README.md:11]: Is this updated to include the SMP rules tested in other PRs?

One minor nit as that old Android devices (API 21) could use a keyboard font 😢

Comment (darcywong00) [reply common/test/keyboards/web_context_tests/README.md:11]: ah, I found that keyboard on #6272

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.204-alpha](https://keyman.com/downloads/releases/alpha/15.0.204)

**Pull request merged**